src/wrappers/python/README.md

```diff
@@ -69,30 +66,94 @@ for more information.
-    h['channels'] = {'R' : Imath.Channel(FLOAT),
-                     'G' : Imath.Channel(FLOAT),
-                     'B' : Imath.Channel(FLOAT),
-                     'A' : Imath.Channel(FLOAT)} 
-    o = OpenEXR.OutputFile("hello.exr", h)
-    r = array('f', [n for n in range(size*0,size*1)]).tobytes()
-    g = array('f', [n for n in range(size*1,size*2)]).tobytes()
-    b = array('f', [n for n in range(size*2,size*3)]).tobytes()
-    a = array('f', [n for n in range(size*3,size*4)]).tobytes()
-    channels = {'R' : r, 'G' : g, 'B' : b, 'A' : a}
-    o.writePixels(channels)
-    o.close()
+The "Hello, World" image writer:
+
+    # Generate arrays for R, G, and B channels with random values
+    height, width = (20, 10)
+    R = np.random.rand(height, width).astype('f')
+    G = np.random.rand(height, width).astype('f')
+    B = np.random.rand(height, width).astype('f')
+    channels = { "R" : R, "G" : G, "B" : B }
+    header = { "compression" : OpenEXR.ZIP_COMPRESSION,
+               "type" : OpenEXR.scanlineimage }
+
+    with OpenEXR.File(header, channels) as outfile:
+        outfile.write("readme.exr")
+
+Or alternatively, construct the same output file via a single RGB pixel array:
+
+    # Generate a 3D NumPy array for RGB channels with random values
+    height, width = (20, 10)
+    RGB = np.random.rand(height, width, 3).astype('f')
+
+    channels = { "RGB" : RGB }
+    header = { "compression" : OpenEXR.ZIP_COMPRESSION,
+               "type" : OpenEXR.scanlineimage }
+
+    with OpenEXR.File(header, channels) as outfile:
+        outfile.write("readme.exr")
+
+The corresponding example of reading an image is:
+
+    with OpenEXR.File("readme.exr") as infile:
+
+        header = infile.header()
+        print(f"type={header['type']}")
+        print(f"compression={header['compression']}")
+
+        R = infile.channels()["R"].pixels
+        G = infile.channels()["G"].pixels
+        B = infile.channels()["B"].pixels
+        height, width = R.shape
+        for y in range(height):
+            for x in range(width):
+                pixel = (R[y, x], G[y, x], B[y, x])
+                print(f"pixel[{y}][{x}]={pixel}")
+
+Or alternatively, read the data as a single RGBA array:
+
+    with OpenEXR.File("readme.exr", rgba=True) as infile:
+
+        RGB = infile.channels()["RGB"].pixels
+        height, width, _ = RGB.shape
+        for y in range(height):
+            for x in range(width):
+                pixel = tuple(RGB[y, x])
+                print(f"pixel[{y}][{x}]={pixel}")
+
+To modify the header metadata in a file:
+
+    with OpenEXR.File("readme.exr") as f:
+        
+        f.header()["displayWindow"] = OpenEXR.Box2i((3,4),(5,6))
+        f.header()["comments"] = "test image"
+        f.header()["longitude"] = -122.5
+        f.write("readme_modified.exr")
+
+        with OpenEXR.File("readme_modified.exr") as o:
+            assert o.header()["displayWindow"] == OpenEXR.Box2i((3,4),(5,6))
```

Comment:
If we want the outward facing API to be really pythonic, probably we don't need the IMath types?

A more pythonic way would simply be `o.header()["displayWindow"] == ((3 ,4), (5, 6))`

I looked up `src\wrappers\python\IMath.py`, it doesn't seem there is a substantial difference from python tuples, except for the print (repr) method. `IMath.Box` did not really type it all the way down, with its `min` and `max` looks like python tuples (although the author's intention is to use `IMath.point`).

If we really want to enforce the dimensionality of the parameter but remain pythonic or num-pythonic, maybe set this to an `ndarray` and assert its shape is `(2, 2)`. I tend to think a few classes (esp. `point` and `box`) in `Imath.py` is reinventing the wheel in a C-style and doesn't really read like modern numpy/scipy code. These data could all just be ndarrays, that's actually less confusion than letting the user figure out whether `box.min` should be plain tuple or `Imath.point`. The important things are this parameter's type and shape, which are all enforced in ndarry for free (up on assign, arithmetic and comparison).

what do people think?

Reply:
I'm also inclined to believe that maybe we should accept and promote the built-in Python types (tuples, arrays, dictionaries) for all the usual "short" data types for which they seem a decent fit, and use Numpy arrays for blocks of image pixel data, and call it a day. The Imath types are not Pythonic, but rather look like somebody simply wrapped a C++ library. They're not friendly to people who must pass data back and forth between OpenEXR and almost any other Python library (Alembic is the only Python interface I know of that expects people to use classes from Imath's Python bindings).